pkgs/build-support/go/module.nix

```
@@ -145,7 +149,6 @@ let
 
     outputHashMode = "recursive";
     outputHash = vendorHash;
-    outputHashAlgo = if args' ? vendorSha256 || vendorHash == "" then "sha256" else null;
```

```nix
outputHashAlgo = if vendorHash == "" then "sha256" else null;
```
Need to keep this to avoid `error: empty hash requires explicit hash algorithm`.

---

Maybe

```suggestion
    outputHashAlgo = lib.optionalString (vendorHash == "") (lib.head (lib.splitString lib.fakeHash));
```

BTW, it looks like specifying `outputHashAlgo = null;` and `outputHashAlgo = "";` are semantically the same, and both version appears in Nixpkgs. Here I choose "" when `vendorHash` for simpler implementation and better type consistency.

---

Addressed.

---

Please use my version, I think it is simpler and it is the same as is used elsewhere, e.g.:

https://github.com/NixOS/nixpkgs/blob/3b8cd8ad708f72f237daf05b42a956184bf2996d/pkgs/build-support/rust/fetch-cargo-tarball/default.nix#L33

---

Okay. Applied.